uPortal-api/uPortal-api-rest/src/main/java/org/apereo/portal/json/rendering/JsonLayoutRenderingPipeline.java

```diff
@@ -75,27 +75,28 @@ public void renderState(HttpServletRequest req, HttpServletResponse res)
                 this.pipeline.getEventReader(req, res);
 
         // set the response mime type
+
         final String contentType = "application/json; charset=" + CHARACTER_SET;
         res.setContentType(contentType);
 
         final PrintWriter writer = res.getWriter();
-
-        for (final CharacterEvent event : pipelineEventReader) {
-            if (CharacterEventTypes.CHARACTER != event.getEventType()) {
-                throw new RenderingPipelineConfigurationException(
-                        "Only "
-                                + CharacterEventTypes.CHARACTER
-                                + " events are supported in the top level renderer. "
-                                + event.getEventType()
-                                + " is not supported.");
+        if (pipelineEventReader != null) {
+            for (final CharacterEvent event : pipelineEventReader) {
+                if (CharacterEventTypes.CHARACTER != event.getEventType()) {
+                    throw new RenderingPipelineConfigurationException(
+                            "Only "
+                                    + CharacterEventTypes.CHARACTER
+                                    + " events are supported in the top level renderer. "
+                                    + event.getEventType()
+                                    + " is not supported.");
+                }
+
+                final String data = ((CharacterDataEvent) event).getData();
+                writer.print(data);
+                writer.flush();
+                res.flushBuffer();
             }
```

Comment:
From a design perspective, should there always be a pipelineEventReader? If so, would throwing an IllegalStateException be in order, rather than returning null? Suggest at least an warn in the logs.